src/System.Net.Sockets/src/System/Net/Sockets/AcceptOverlappedAsyncResult.Windows.cs

```diff
@@ -101,8 +96,6 @@ internal void SetUnmanagedStructures(byte[] buffer, int addressBufferLength)
 
         private void LogBuffer(long size)
         {
-            if (!NetEventSource.IsEnabled) return;
```

Comment:
I guess I could add the asserts with a comment explaining it's racy... most likely any hits would be for real violations, and if it ever hit due to the race, it could be removed.